Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.5.1-all.zip'.

**Describe the bug**

```
./gradlew run
````

produced the following error

```
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:52)
        at io.quarkus.runtime.Quarkus.launchFromIDE(Quarkus.java:93)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at com.beust.cedlinks.MainKt.main(Main.kt:15)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:49)
        ... 5 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.5.1-all.zip'.
```

**Expected behavior**

The application should start.

**Actual behavior**

The application doesn't start.


**To Reproduce**

Using Gradle 6.5.1 on Windows 10. Haven't uploaded this project to github yet but can do if necessary.

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.datasource.db-kind=h2
quarkus.datasource.username=sa
quarkus.datasource.password=password
quarkus.datasource.jdbc.url=jdbc:h2:mem:testdb

quarkus.http.port=8080
```

**Environment (please complete the following information):**

Windows 10

```
$ java -version
java version "14.0.2" 2020-07-14
Java(TM) SE Runtime Environment (build 14.0.2+12-46)
Java HotSpot(TM) 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)
```

```
$ ./gradlew --version

------------------------------------------------------------
Gradle 6.5.1
------------------------------------------------------------

Build time:   2020-06-30 06:32:47 UTC
Revision:     66bc713f7169626a7f0134bf452abde51550ea0a

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          14.0.2 (Oracle Corporation 14.0.2+12-46)
OS:           Windows 10 10.0 amd64
```

**Additional context**

Full stack trace

```
59:51,043 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
13:59:51,043 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
13:59:51,044 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [io.micronaut.http] to DEBUG
13:59:51,044 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
13:59:51,045 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@65b3f4a4 - Registering current configuration as safe fallback point

SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/Cedric/.gradle/caches/modules-2/files-2.1/org.jboss.slf4j/slf4j-jboss-logging/1.2.0.Final/bff294c02b64ad6bf8af6e6994e186dc035e0a47/slf4j-jboss-logging-1.2.0.Final.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/Cedric/.gradle/caches/modules-2/files-2.1/ch.qos.logback/logback-classic/1.2.3/7c4f3c474fb2c041d8028740440937705ebb473a/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.jboss.slf4j.JBossLoggerFactory]
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:52)
        at io.quarkus.runtime.Quarkus.launchFromIDE(Quarkus.java:93)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at com.beust.cedlinks.MainKt.main(Main.kt:15)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:49)
        ... 5 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.5.1-all.zip'.
        at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:58)
        ... 10 more
Caused by: java.lang.RuntimeException: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.5.1-all.zip'.
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:135)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:85)
        at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:56)
        ... 10 more
Caused by: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.5.1-all.zip'.
        at org.gradle.tooling.internal.consumer.loader.DefaultToolingImplementationLoader.create(DefaultToolingImplementationLoader.java:102)
        at org.gradle.tooling.internal.consumer.loader.CachingToolingImplementationLoader.create(CachingToolingImplementationLoader.java:45)
        at org.gradle.tooling.internal.consumer.loader.SynchronizedToolingImplementationLoader.create(SynchronizedToolingImplementationLoader.java:44)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.onStartAction(LazyConsumerActionExecutor.java:147)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:129)
        at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
        at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61)
        at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
        at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.lambda$run$0(DefaultAsyncConsumerActionExecutor.java:62)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:832)
        at org.gradle.tooling.internal.consumer.BlockingResultHandler.getResult(BlockingResultHandler.java:46)
        at org.gradle.tooling.internal.consumer.DefaultModelBuilder.get(DefaultModelBuilder.java:51)
        at io.quarkus.bootstrap.resolver.QuarkusGradleModelFactory.createForTasks(QuarkusGradleModelFactory.java:25)
        at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:47)
        at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:24)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:132)
        ... 12 more
Caused by: java.lang.RuntimeException: Could not create an implementation of service 'org.gradle.tooling.internal.protocol.ConnectionVersion4'.
        at org.gradle.internal.service.DefaultServiceLocator$ServiceFactory.newInstance(DefaultServiceLocator.java:195)
        at org.gradle.internal.service.DefaultServiceLocator$ServiceFactory.create(DefaultServiceLocator.java:188)
        at org.gradle.tooling.internal.consumer.loader.DefaultToolingImplementationLoader.create(DefaultToolingImplementationLoader.java:82)
        at org.gradle.tooling.internal.consumer.loader.CachingToolingImplementationLoader.create(CachingToolingImplementationLoader.java:45)
        at org.gradle.tooling.internal.consumer.loader.SynchronizedToolingImplementationLoader.create(SynchronizedToolingImplementationLoader.java:44)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.onStartAction(LazyConsumerActionExecutor.java:147)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:129)
        at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
        at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61)
        at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
        at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.lambda$run$0(DefaultAsyncConsumerActionExecutor.java:62)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: org.gradle.api.reflect.ObjectInstantiationException: Could not create an instance of type org.gradle.tooling.internal.provider.DefaultConnection.
        at org.gradle.internal.reflect.DirectInstantiator.newInstance(DirectInstantiator.java:66)
        at org.gradle.internal.reflect.DirectInstantiator.instantiate(DirectInstantiator.java:45)
        at org.gradle.internal.service.DefaultServiceLocator$ServiceFactory.newInstance(DefaultServiceLocator.java:193)
        ... 16 more
Caused by: java.lang.LinkageError: loader constraint violation in interface itable initialization for class org.gradle.tooling.internal.provider.DefaultConnection: when selecting method 'org.gradle.tooling.internal.protocol.ConnectionMetaDataVersion1 org.gradle.tooling.internal.protocol.ConnectionVersion4.getMetaData()' the class loader io.quarkus.launcher.QuarkusLauncher$IDEClassLoader @282003e1 for super interface org.gradle.tooling.internal.protocol.ConnectionVersion4, and the class loader org.gradle.internal.classloader.VisitableURLClassLoader @718d58e of the selected method's class, org.gradle.tooling.internal.provider.DefaultConnection have different Class objects for the type org.gradle.tooling.internal.protocol.ConnectionMetaDataVersion1 used in the signature (org.gradle.tooling.internal.protocol.ConnectionVersion4 is in unnamed module of loader io.quarkus.launcher.QuarkusLauncher$IDEClassLoader @282003e1, parent loader 'app'; org.gradle.tooling.internal.provider.DefaultConnection is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @718d58e, parent loader org.gradle.internal.classloader.FilteringClassLoader @412267b0)
        at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3215)
        at java.base/java.lang.Class.getConstructors(Class.java:1957)
        at org.gradle.internal.reflect.DirectInstantiator$ConstructorCache.create(DirectInstantiator.java:91)
        at org.gradle.internal.reflect.DirectInstantiator$ConstructorCache.create(DirectInstantiator.java:86)
        at org.gradle.internal.reflect.ReflectionCache$WeaklyClassReferencingCache.get(ReflectionCache.java:56)
        at org.gradle.internal.reflect.ReflectionCache.get(ReflectionCache.java:36)
        at org.gradle.internal.reflect.DirectInstantiator.newInstance(DirectInstantiator.java:60)
        ... 18 more

> Task :run FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command 'C:\Program Files\Java\jdk-14.0.2\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.5.1/userguide/command_line_interface.html#sec:command_line_warnings
```

FYI - I get the same behavior with gradle 6. Repro on win 10, jdk-14.0.1.7, with basically the template project created on quarkus.io.

Full stack trace

```
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:52)
        at io.quarkus.runtime.Quarkus.launchFromIDE(Quarkus.java:93)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at org.maclei.JavaMain.main(JavaMain.java:11)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:49)
        ... 4 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.6-bin.zip'.
        at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:58)
        ... 9 more
Caused by: java.lang.RuntimeException: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.6-bin.zip'.
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:135)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:85)
        at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:56)
        ... 9 more
Caused by: org.gradle.tooling.GradleConnectionException: Could not create an instance of Tooling API implementation using the specified Gradle distribution 'https://services.gradle.org/distributions/gradle-6.6-bin.zip'.
        at org.gradle.tooling.internal.consumer.loader.DefaultToolingImplementationLoader.create(DefaultToolingImplementationLoader.java:102)       
        at org.gradle.tooling.internal.consumer.loader.CachingToolingImplementationLoader.create(CachingToolingImplementationLoader.java:45)        
        at org.gradle.tooling.internal.consumer.loader.SynchronizedToolingImplementationLoader.create(SynchronizedToolingImplementationLoader.java:44)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.onStartAction(LazyConsumerActionExecutor.java:147)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:129)
        at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
        at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61) 
        at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
        at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.lambda$run$0(DefaultAsyncConsumerActionExecutor.java:62)   
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:832)
        at org.gradle.tooling.internal.consumer.BlockingResultHandler.getResult(BlockingResultHandler.java:46)
        at org.gradle.tooling.internal.consumer.DefaultModelBuilder.get(DefaultModelBuilder.java:51)
        at io.quarkus.bootstrap.resolver.QuarkusGradleModelFactory.createForTasks(QuarkusGradleModelFactory.java:25)
        at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:47)
        at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:24)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:132)
        ... 11 more
Caused by: java.lang.RuntimeException: Could not create an implementation of service 'org.gradle.tooling.internal.protocol.ConnectionVersion4'.     
        at org.gradle.internal.service.DefaultServiceLocator$ServiceFactory.newInstance(DefaultServiceLocator.java:195)
        at org.gradle.internal.service.DefaultServiceLocator$ServiceFactory.create(DefaultServiceLocator.java:188)
        at org.gradle.tooling.internal.consumer.loader.DefaultToolingImplementationLoader.create(DefaultToolingImplementationLoader.java:82)        
        at org.gradle.tooling.internal.consumer.loader.CachingToolingImplementationLoader.create(CachingToolingImplementationLoader.java:45)        
        at org.gradle.tooling.internal.consumer.loader.SynchronizedToolingImplementationLoader.create(SynchronizedToolingImplementationLoader.java:44)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.onStartAction(LazyConsumerActionExecutor.java:147)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:129)
        at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
        at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61) 
        at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
        at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.lambda$run$0(DefaultAsyncConsumerActionExecutor.java:62)   
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: org.gradle.api.reflect.ObjectInstantiationException: Could not create an instance of type org.gradle.tooling.internal.provider.DefaultConnection.
        at org.gradle.internal.reflect.DirectInstantiator.newInstance(DirectInstantiator.java:66)
        at org.gradle.internal.reflect.DirectInstantiator.instantiate(DirectInstantiator.java:45)
        at org.gradle.internal.service.DefaultServiceLocator$ServiceFactory.newInstance(DefaultServiceLocator.java:193)
        ... 16 more
Caused by: java.lang.LinkageError: loader constraint violation in interface itable initialization for class org.gradle.tooling.internal.provider.DefaultConnection: when selecting method 'org.gradle.tooling.internal.protocol.ConnectionMetaDataVersion1 org.gradle.tooling.internal.protocol.ConnectionVersion4.getMetaData()' the class loader io.quarkus.launcher.QuarkusLauncher$IDEClassLoader @2ac273d3 for super interface org.gradle.tooling.internal.protocol.ConnectionVersion4, and the class loader org.gradle.internal.classloader.VisitableURLClassLoader @642b9143 of the selected method's class, org.gradle.tooling.internal.provider.DefaultConnection have different Class objects for the type org.gradle.tooling.internal.protocol.ConnectionMetaDataVersion1 used in the signature (org.gradle.tooling.internal.protocol.ConnectionVersion4 is in unnamed module of loader io.quarkus.launcher.QuarkusLauncher$IDEClassLoader @2ac273d3, parent loader 'app'; org.gradle.tooling.internal.provider.DefaultConnection is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @642b9143, parent loader org.gradle.internal.classloader.FilteringClassLoader @404257b9)    
        at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3215)
        at java.base/java.lang.Class.getConstructors(Class.java:1957)
        at org.gradle.internal.reflect.DirectInstantiator$ConstructorCache.create(DirectInstantiator.java:91)
        at org.gradle.internal.reflect.DirectInstantiator$ConstructorCache.create(DirectInstantiator.java:86)
        at org.gradle.internal.reflect.ReflectionCache$WeaklyClassReferencingCache.get(ReflectionCache.java:56)
        at org.gradle.internal.reflect.ReflectionCache.get(ReflectionCache.java:36)
        at org.gradle.internal.reflect.DirectInstantiator.newInstance(DirectInstantiator.java:60)
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/IDEDevModeMain.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/IDEDevModeMain.java
@@ -14,7 +14,6 @@ import io.quarkus.bootstrap.BootstrapGradleException;
 import io.quarkus.bootstrap.app.CuratedApplication;
 import io.quarkus.bootstrap.model.AppArtifactKey;
 import io.quarkus.bootstrap.resolver.AppModelResolverException;
-import io.quarkus.bootstrap.resolver.QuarkusGradleModelFactory;
 import io.quarkus.bootstrap.resolver.maven.workspace.LocalProject;
 import io.quarkus.bootstrap.resolver.model.QuarkusModel;
 import io.quarkus.bootstrap.resolver.model.WorkspaceModule;
@@ -43,18 +42,17 @@ public class IDEDevModeMain implements BiConsumer<CuratedApplication, Map<String
                     devModeContext.getAdditionalModules().add(toModule(module.getValue()));
                 }
             } else {
-                // TODO find a way to reuse the previously model instead of building a new one.
-                QuarkusModel quarkusModel = QuarkusGradleModelFactory.createForTasks(
-                        BuildToolHelper.getBuildFile(appClasses, BuildToolHelper.BuildTool.GRADLE).toFile(),
-                        QuarkusModelHelper.DEVMODE_REQUIRED_TASKS);
-                final WorkspaceModule launchingModule = quarkusModel.getWorkspace().getMainModule();
-                DevModeContext.ModuleInfo root = toModule(quarkusModel.getWorkspace().getMainModule());
+                final QuarkusModel model = QuarkusModelHelper
+                        .deserializeQuarkusModel((Path) stringObjectMap.get(QuarkusModelHelper.SERIALIZED_QUARKUS_MODEL));
+                final WorkspaceModule launchingModule = model.getWorkspace().getMainModule();
+                DevModeContext.ModuleInfo root = toModule(launchingModule);
                 devModeContext.setApplicationRoot(root);
-                for (WorkspaceModule additionalModule : quarkusModel.getWorkspace().getAllModules()) {
+                for (WorkspaceModule additionalModule : model.getWorkspace().getAllModules()) {
                     if (!additionalModule.getArtifactCoords().equals(launchingModule.getArtifactCoords())) {
                         devModeContext.getAdditionalModules().add(toModule(additionalModule));
                     }
                 }
+
             }
 
         } catch (AppModelResolverException e) {

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/IDELauncherImpl.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/IDELauncherImpl.java
@@ -32,10 +32,12 @@ public class IDELauncherImpl {
 
             if (!BuildToolHelper.isMavenProject(projectRoot)) {
                 final QuarkusModel quarkusModel = BuildToolHelper.enableGradleAppModelForDevMode(projectRoot);
-                // Gradle uses a different output directory for classes, we override the one used by the IDE
+                context.put(QuarkusModelHelper.SERIALIZED_QUARKUS_MODEL,
+                        QuarkusModelHelper.serializeQuarkusModel(quarkusModel));
+
                 final WorkspaceModule launchingModule = quarkusModel.getWorkspace().getMainModule();
                 Path launchingModulePath = QuarkusModelHelper.getClassPath(launchingModule);
-
+                // Gradle uses a different output directory for classes, we override the one used by the IDE
                 builder.setProjectRoot(launchingModulePath)
                         .setApplicationRoot(launchingModulePath);
 

--- a/independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/util/QuarkusModelHelper.java
+++ b/independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/util/QuarkusModelHelper.java
@@ -15,6 +15,8 @@ import io.quarkus.bootstrap.resolver.model.WorkspaceModule;
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 import java.nio.file.FileSystem;
 import java.nio.file.FileSystems;
@@ -35,6 +37,7 @@ public class QuarkusModelHelper {
 
     }
 
+    public final static String SERIALIZED_QUARKUS_MODEL = "quarkus-internal.serialized-quarkus-model.path";
     public final static String[] DEVMODE_REQUIRED_TASKS = new String[] { "classes" };
     public final static String[] TEST_REQUIRED_TASKS = new String[] { "classes", "testClasses" };
 
@@ -54,6 +57,28 @@ public class QuarkusModelHelper {
             out.writeObject(QuarkusModelHelper.convert(model, appArtifact));
         }
         return serializedModel;
+    }
+
+    public static Path serializeQuarkusModel(QuarkusModel model) throws IOException {
+        final Path serializedModel = File.createTempFile("quarkus-model", ".dat").toPath();
+        try (ObjectOutputStream out = new ObjectOutputStream(Files.newOutputStream(serializedModel))) {
+            out.writeObject(model);
+        }
+        return serializedModel;
+    }
+
+    public static QuarkusModel deserializeQuarkusModel(Path modelPath) throws BootstrapGradleException {
+        if (Files.exists(modelPath)) {
+            try (InputStream existing = Files.newInputStream(modelPath);
+                    ObjectInputStream object = new ObjectInputStream(existing)) {
+                QuarkusModel model = (QuarkusModel) object.readObject();
+                IoUtils.recursiveDelete(modelPath);
+                return model;
+            } catch (IOException | ClassNotFoundException e) {
+                throw new BootstrapGradleException("Failed to deserialize quarkus model", e);
+            }
+        }
+        throw new BootstrapGradleException("Unable to locate quarkus model");
     }
 
     public static Path getClassPath(WorkspaceModule model) throws BootstrapGradleException {

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaApplicationModuleDevModeTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaApplicationModuleDevModeTest.java
@@ -1,0 +1,34 @@
+package io.quarkus.gradle.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.UUID;
+
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+
+import com.google.common.collect.ImmutableMap;
+
+@DisabledOnOs(OS.WINDOWS)
+public class BasicJavaApplicationModuleDevModeTest extends QuarkusDevGradleTestBase {
+
+    @Override
+    protected String projectDirectoryName() {
+        return "basic-java-application-project";
+    }
+
+    @Override
+    protected String[] buildArguments() {
+        return new String[] { "clean", "run", "-s" };
+    }
+
+    protected void testDevMode() throws Exception {
+        assertThat(getHttpResponse("/hello")).contains("hello");
+
+        final String uuid = UUID.randomUUID().toString();
+        replace("src/main/java/org/acme/ExampleResource.java",
+                ImmutableMap.of("return \"hello\";", "return \"" + uuid + "\";"));
+
+        assertUpdatedResponseContains("/hello", uuid);
+    }
+}

--- a/integration-tests/gradle/src/test/resources/basic-java-application-project/build.gradle
+++ b/integration-tests/gradle/src/test/resources/basic-java-application-project/build.gradle
@@ -1,0 +1,26 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+    id 'application'
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+}
+
+dependencies {
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'io.quarkus:quarkus-resteasy'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured'
+}
+
+compileJava {
+    options.compilerArgs << '-parameters'
+}
+
+application {
+    mainClass = 'org.acme.EntryPoint'
+}

--- a/integration-tests/gradle/src/test/resources/basic-java-application-project/gradle.properties
+++ b/integration-tests/gradle/src/test/resources/basic-java-application-project/gradle.properties
@@ -1,0 +1,2 @@
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformGroupId=io.quarkus

--- a/integration-tests/gradle/src/test/resources/basic-java-application-project/settings.gradle
+++ b/integration-tests/gradle/src/test/resources/basic-java-application-project/settings.gradle
@@ -1,0 +1,14 @@
+pluginManagement {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    plugins {
+      id 'io.quarkus' version "${quarkusPluginVersion}"
+    }
+}
+rootProject.name='code-with-quarkus'
+include 'library'
+include 'application'
+

--- a/integration-tests/gradle/src/test/resources/basic-java-application-project/src/main/java/org/acme/EntryPoint.java
+++ b/integration-tests/gradle/src/test/resources/basic-java-application-project/src/main/java/org/acme/EntryPoint.java
@@ -1,0 +1,11 @@
+package org.acme;
+
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.runtime.annotations.QuarkusMain;
+
+@QuarkusMain
+public class EntryPoint {
+    public static void main(String[] args) {
+        Quarkus.run();
+    }
+}

--- a/integration-tests/gradle/src/test/resources/basic-java-application-project/src/main/java/org/acme/ExampleResource.java
+++ b/integration-tests/gradle/src/test/resources/basic-java-application-project/src/main/java/org/acme/ExampleResource.java
@@ -1,0 +1,16 @@
+package org.acme;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class ExampleResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+}

